src/main/java/io/github/fusionflux/portalcubed/content/fizzler/DisintegrationSoundType.java

```diff
@@ -0,0 +1,40 @@
+package io.github.fusionflux.portalcubed.content.fizzler;
+
+import com.google.common.collect.Iterables;
+
+import io.github.fusionflux.portalcubed.PortalCubed;
+import net.minecraft.core.Registry;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.sounds.SoundEvent;
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.entity.EntityType;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+public enum DisintegrationSoundType {
+	GENERIC,
+	RADIO,
+	TURRET;
+
+	public final String name = this.name().toLowerCase(Locale.ROOT);
+	public final SoundEvent sound;
+	public final TagKey<EntityType<?>> tag;
+
+	DisintegrationSoundType() {
+		ResourceLocation id = PortalCubed.id(name + "_disintegration");
+		this.sound = Registry.register(BuiltInRegistries.SOUND_EVENT, id, SoundEvent.createVariableRangeEvent(id));
```

Comment:
sound events should be in a PortalCubedSounds and then referenced in constructors

Reply:
similar thing for tags

Reply:
I don't really like this plus you didn't have any problems with the prop impact sound registering the sound event in its enum constructor

Reply:
I didn't see that

Reply:
I guess it's fine